sklearn/metrics/_classification.py

```diff
@@ -655,10 +665,11 @@ def multilabel_confusion_matrix(
     """
     y_true, y_pred = attach_unique(y_true, y_pred)
     xp, _, device_ = get_namespace_and_device(y_true, y_pred)
-    y_type, y_true, y_pred = _check_targets(y_true, y_pred)
+    y_type, y_true, y_pred, sample_weight = _check_targets(
+        y_true, y_pred, sample_weight
+    )
     if sample_weight is not None:
         sample_weight = column_or_1d(sample_weight, device=device_)
```

Comment:
We keep that for the device, right ?
It feels a bit weird. Do you think `check_targets` should be responsible for this ?